study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/UpdateActivityNamingDetails.java

```diff
@@ -0,0 +1,143 @@
+package org.broadinstitute.ddp.studybuilder.task;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Functions;
+import com.typesafe.config.Config;
+import org.broadinstitute.ddp.db.dao.ActivityI18nDao;
+import org.broadinstitute.ddp.db.dao.JdbiUmbrellaStudy;
+import org.broadinstitute.ddp.db.dao.UserDao;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.model.activity.definition.i18n.ActivityI18nDetail;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.studybuilder.ActivityBuilder;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * General task to help update activity naming details such as name, title, subtitle, and description. The activity
+ * definition will be read to lookup the latest text to update to. If new text in a new language is added, those will be
+ * inserted as well. This task will do this for all activities in the study.
+ */
+public class UpdateActivityNamingDetails implements CustomTask {
```

Comment:
These naming details are not revisioned, so it was straightforward to write a "general" task like this one.